packages/create-cloudflare/src/helpers/command.ts

```diff
@@ -185,11 +186,19 @@ export const retry = async <T>(
 	throw error;
 };
 
-// Prints the section header & footer while running the `cmd`
 export const runFrameworkGenerator = async (
 	ctx: PagesGeneratorContext,
-	cmd: string
+	argString: string
 ) => {
+	const cli = getFrameworkCli(ctx);
+	const { npm, dlx } = detectPackageManager();
+
+	// yarn cannot `yarn create@some-version` and doesn't have an npx equivalent
+	// So to retain the ability to lock verisons we run it with `npx` and spoof
+	// the user agent so scaffolding tools treat the invocation like yarn
+	let cmd = `${npm === "yarn" ? "npx" : dlx} ${cli} ${argString}`;
+	const env = npm === "yarn" ? { npm_config_user_agent: "yarn" } : {};
```

Comment:
why is this needed?

Reply:
So that the scaffolding tools think that they were invoked via `yarn create my-framework`. This means that they'll use `yarn` when doing any npm installs, which in turn means that we don't need to `resetPackageManager`.

Reply:
ah ok, so here is basically when we run `npx` instead of yarn and we add this user agent to trick the underlying CLIs to think that we're actually using yarn?

Reply:
exactly!

Reply:
I assume the code comments were added after this discussion?